src/cpp/rtps/DataSharing/ReaderPool.hpp

```diff
@@ -67,7 +64,10 @@ class ReaderPool : public DataSharingPayloadPool
             return true;
         }
 
-        return false;
+        // If owner is not this, then it must be an intraprocess datasharing writer
```

Comment:
Could we `assert` this condition?

```suggestion
        // If owner is not this, then it must be an intraprocess datasharing writer
        assert(nullptr != dynamic_cast<WriterPool>(data_owner));
```